### ISSUE TITLE
Log started, stop actor system on binding failure operation on server builder

### ISSUE THREAD

**Issue by johanandren** [akka/akka-http#3758]
A very common snippet is something like this (copied from the Akka microservice guide): 

```
val bound = Http()
        .newServerAt(interface, port)
        .bind(service)
        .map(_.addToCoordinatedShutdown(3.seconds)) 

 bound.onComplete { // <3>
      case Success(binding) =>
        val address = binding.localAddress
        system.log.info(
          "Shopping online at gRPC server {}:{}",
          address.getHostString,
          address.getPort)
      case Failure(ex) =>
        system.log.error("Failed to bind gRPC endpoint, terminating system", ex)
        system.terminate()
    }
```

Could we just build that into the builder somehow (perhaps including the coordinated shutdown hookup)?

```
Http()
        .newServerAt(interface, port)
        .logStarted()
        .withDefaultTerminationHandling(shutdownTimeout = 3 seconds)
        .bind(service)
```

**Comment by jrudolph** [comment]
Yes, that totally makes sense, I hinted at such a feature in #3389 and #3390. Good to collect some more concrete ideas how it would look like.

**Comment by johanandren** [comment]
Ups accidental-closed but perhaps this is pretty much a duplicate of #3389 anyway.

**Comment by octonato** [comment]
I'm thinking on how we could make it easier in the context of using it together with cluster. 

In the shopping cart sample, we have setup the project to be able to run 3 nodes on a local machine. Each node will pick a different http port, eg: 8101, 8102 and 8203.

The interface and port are defined in different config files. 

Because of that, we must first read the configs and then pass it to `newServerAt`. 

I think it would be convenient if define a default port (not 80) and use it as follow.

```hocon
akka.http.interface = "0.0.0.0"
akka.http.port = 8080
```

```scala
Http().newServer().bind(service)
```

Users could of course still use `newServerAt`.

**Comment by octonato** [comment]
About `logStarted()`, why not enable by default and have `disableStartupLog()`

And `withDefaultTerminationHandling(shutdownTimeout = 3 seconds)`, why not have it defined by default and have `withTerminationHandling(shutdownTimeout = 3 seconds)`.

Then
```scala
Http().newServer().bind(service)
```
would have good defaults.

We could even go one step further and say:

```scala
Http().newServer(service)
```

**Comment by jrudolph** [comment]
@octonato 

> akka.http.interface = "0.0.0.0"
> akka.http.port = 8080

I agree that would be useful in general. I find that I'm reimplementing copy-and-pasting that part in every serious project anyways.

We probably considered something like this when introducing the new ServerBuilder API where we wanted to avoid more overload madness.

Maybe we could introduce

```
Http().newServerFromConfig().bind()
```

and

```
Http().newServerFromSettings(hostSetting, portSetting).bind()
```

WDYT?

Graceful termination isn't completely free and it's also not clear what a good default should be. In any case, we should make it possible to enable it with a setting.
